toolkit/tools/grapher/grapher.go

```diff
@@ -205,6 +208,36 @@ func addNodesForPackage(g *pkggraph.PkgGraph, pkg *pkgjson.Package) (foundDuplic
 	return
 }
 
+// handleRemoteDependency ensures that a remote node is available in the graph for every unresolved dependency.
+// 1. Check if the exact dependency is already in the graph. If it is, reuse it.
+// 2. If it is not, create a new unresolved node for the dependency.
+// It is important that we only match on the exact dependency name and version. If we don't, we may end up with
+// unpredictable behavior in the scheduler. If two different remote dependencies are added to two different build
+// nodes of a single SRPM, then the scheduler may queue that node twice.
+func handleRemoteDependency(g *pkggraph.PkgGraph, dependency *pkgjson.PackageVer) (resolvedNode *pkggraph.PkgNode, err error) {
+	existingRemoteNode, err := g.FindExactPkgNodeFromPkg(dependency)
+	if err != nil {
+		err = fmt.Errorf("failed to check lookup list for exact remote %+v:\n%w", dependency, err)
+		return nil, err
+	}
+
+	if existingRemoteNode == nil {
+		// No exact match, add a new one.
+		resolvedNode, err = addUnresolvedPackage(g, dependency)
```

Comment:
I think the word "resolved" is overloaded in this scope. Could we use something else than `resolvedNode` so we don't confuse it with resolved/unresolved nodes in the graph? If I understand correctly, we are actually creating a new **unresolved** node, so I'd be very careful about the wording.